ranzen/torch/data.py

```diff
@@ -141,12 +140,13 @@ def prop_random_split(
         section_sizes.append(len_ - sum(section_sizes))
     generator = torch.default_generator if seed is None else torch.Generator().manual_seed(seed)
     indices = torch.randperm(sum(section_sizes), generator=generator).tolist()
-    splits = []
-    for offset, length in zip(np.cumsum(section_sizes), section_sizes):
-        split = indices[offset - length : offset]
-        if not as_indices:
-            split = Subset(dataset, indices=split)
-        splits.append(split)
+    splits = [
+        indices[offset - length : offset]
+        for offset, length in zip(np.cumsum(section_sizes), section_sizes)
+    ]
+
+    if not as_indices:
+        return [Subset(dataset, indices=split) for split in splits]
```

Comment:
The new code is probably slower, but if `splits` isn't too large, it shouldn't matter.